pandas/tests/io/test_parquet.py

```diff
@@ -478,6 +480,25 @@ def test_s3_roundtrip(self, df_compat, s3_resource, pa):
         check_round_trip(df_compat, pa,
                          path='s3://pandas-test/pyarrow.parquet')
 
+    def test_partition_cols_supported(self, pa_ge_070, df_full):
+        partition_cols = ['bool', 'int']
+        df = df_full
+        path = tempfile.mkdtemp()
```

Comment:
This can use `with tm.ensure_clean() as path:`. Then you don't have to worry about the `shutil.rmtree` and the imports.

Reply:
tm.ensure_clean() does not create Temporary Directory. It only creates temporary files. Will change the same in another request

Reply:
Silly question, but why are we passing in a directory for `df.to_parquet` ?

Reply:
We are passing the path for the temp directory. pyarrow.write_to_dataset requires a directory to store the partitioned files.

Reply:
@gfyoung For more details, do refer here : https://arrow.apache.org/docs/python/parquet.html#writing-to-partitioned-datasets

Reply:
tempfile.mkdtemp is ok, but pls use this in a with statement so it is cleaned.